### ISSUE TITLE
Performance: Convert YOLOX to TensorRT for 1.5-2.4x inference speedup

### ISSUE THREAD

**Issue by filthyrake** [filthyrake/telescope_cam_detection#157]
## Overview
Convert YOLOX PyTorch model to TensorRT with FP16/INT8 quantization for significant inference speedup.

## Expected Performance Improvement
- **FP16 precision**: 1.5x speedup, <1% accuracy loss
- **INT8 precision**: 2.4x speedup, ~0.8% accuracy loss
- **Impact on our system**: 150-250ms → 60-100ms per frame at 1920x1920 resolution

## Benefits
- Minimal code changes (drop-in replacement)
- Native NVIDIA optimization (kernel fusion, layer optimization)
- Batched inference becomes even more efficient
- No accuracy degradation with FP16

### Phase 1: FP16 Conversion (Low Risk)
```python
import torch_tensorrt

trt_model = torch_tensorrt.compile(
    yolox_model,
    inputs=[torch_tensorrt.Input((1, 3, 1920, 1920))],
    enabled_precisions={torch.float16},
    workspace_size=1 << 30,  # 1GB
)
```

### Phase 2: INT8 Conversion (Requires Calibration)
- Collect 500-1000 representative frames from cameras
- Run PTQ (Post-Training Quantization) calibration
- Validate accuracy on test set
- A/B test against FP16

## Technical Details
- Requires: `torch-tensorrt` package
- Compatible with: NVIDIA A30 GPU (Ampere architecture)
- Storage: ~50% smaller model size with INT8
- Deployment: Save as TorchScript `.ts` file

## Success Metrics
- [ ] FP16: ≥1.4x speedup, ≤1% mAP drop
- [ ] INT8: ≥2.0x speedup, ≤2% mAP drop
- [ ] Batched inference: ≥3x throughput improvement (2 cameras)

## References
- Research: 37-60% FPS increase demonstrated in 2025 studies
- TensorRT docs: https://github.com/pytorch/TensorRT
- INT8 calibration guide: https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/

## Priority
**HIGH** - Biggest performance gain for moderate effort

## Related Issues
- Complements #156 (batching optimization)
- Prerequisite for DeepStream migration (future)
